docs/cunumeric/source/api/indexing.rst

```diff
@@ -9,6 +9,7 @@ Generating index arrays
 .. autosummary::
    :toctree: generated/
 
+   fill_diagonal
```

Comment:
I think this function should be in `manipulation.rst`. Looks like we will need to create a new category, maybe "Modifying array contents".